File: backend/ControleFinanceiro.Api/Models/CreditCardPurchase.cs
namespace ControleFinanceiro.Api.Models
{
    public class CreditCardPurchase
    {
        public int Id { get; set; }
        public required string Description { get; set; }
        public decimal Value { get; set; }
        public DateTime PurchaseDate { get; set; }
        public int NumberInstallments { get; set; } = 1;

        public ICollection<PurchaseInstallment> Installments { get; set; } = new List<PurchaseInstallment>();

        public int CategoryId { get; set; }
        public required Category Category { get; set; }

        public int CreditCardId { get; set; }
        public required CreditCard CreditCard { get; set; }
    }
}

File: frontend/src/types/creditCardPurchase.ts
import type { Category } from "./category";
import type { CreditCard } from "./creditCard";

export type CreditCardPurchase = {
    id: number;
    description: string;
    value: number;
    purchaseDate: string;
    numberInstallments: number;
    categoryId: number;
    category: Category;
    creditCardId: number;
    creditCard: CreditCard;
    installments: PurchaseInstallment[];
}

export type CreditCardPurchaseFormData = {
    description: string;
    value: number;
    purchaseDate: string;
    numberInstallments: number;
    categoryId: number;
    creditCardId: number;
}

export type CreditCardPurchaseFilter = {
    creditCardId?: number;
    categoryId?: number;
}

export type PurchaseInstallment = {
    id: number;
    numberInstallment: number;
    dueDate: string;
    value: number;
    isPaid: boolean;
}